app/containers/Forms/ProductRowIdField.tsx

```diff
@@ -24,7 +24,8 @@ export const ProductRowIdFieldComponent: React.FunctionComponent<Props> = props
       schema: {
         ...props.schema,
         enum: props.query.allProducts.edges.map(({node}) => node.rowId),
-        enumNames: props.query.allProducts.edges.map(({node}) => node.name)
+        enumNames: props.query.allProducts.edges.map(({node}) => node.name),
+        state: props.query.allProducts.edges.map(({node}) => node.state)
```

Comment:
`state` is not part of the json schema definition, and should not be added here. 
Perhaps if the `formData` prop is not the id of an active product, this component could simply render a disabled text input instead of the `StringField`.
This way you woudn't have to tweak how the search dropdown works